### PR TITLE
 [DinoMod] Mutation Trees

### PR DESCRIPTION
**Pull request by Maleclypse**
#### Summary

SUMMARY:  Mods "DinoMod Mutation Trees"

#### Purpose of change
I told @LyleSY  I'd try and make some mutation trees and mutant npcs for DinoMod.  

#### Describe the solution

I've added categories to existing mutations and made a couple new to make some special Dinomod mutations.  I'll add mutatn NPC's once we are content with the mutation trees. Update dreams as well.

#### Describe alternatives you've considered

Breaking my word and becoming a broken man.

#### Testing
Testing Succesful

### CODE REVIEW

**Comment by actual-nh** [comment]
The Travis build is erroring on some JSON problems, BTW.

**Comment by Maleclypse** [comment]
> The Travis build is erroring on some JSON problems, BTW.

Yep, just been busy and couldn't get to it.  Thanks!

**Comment by Maleclypse** [comment]
@LyleSY let me know if you want any changes to mutations/trees and if those are good I'll make random NPCs that can appear with these mutations, assuming you'd like that.

File: data/mods/DinoMod/items/mutagen.json [hunk 1]
```diff
@@ -0,0 +1,44 @@
+[
+  {
+    "id": "iv_mutagen_tyrant",
+    "copy-from": "iv_mutagen_flavor",
+    "type": "COMESTIBLE",
+    "name": "tyrant serum",
+    "description": "A super-concentrated mutagen strongly resembling a children's slime.  You need a syringe to inject it… if you really want to?",
```
Comment (actual-nh) [inline data/mods/DinoMod/items/mutagen.json:7]:
```suggestion
    "description": "A super-concentrated mutagen strongly resembling children's toy slime.  You need a syringe to inject it… if you really want to?",
```

File: data/mods/DinoMod/items/mutagen.json [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    "color": "light_green",
+    "healthy": -4,
+    "use_action": { "type": "mutagen_iv", "mutation_category": "  TYRANT" }
+  },
+  {
+    "id": "mutagen_tyrant",
+    "copy-from": "mutagen_flavor",
+    "type": "COMESTIBLE",
+    "name": "tyrant mutagen",
+    "description": "An extremely rare mutagen cocktail, it smells like blood and is a green goop with blood red streaks",
```
Comment (actual-nh) [inline data/mods/DinoMod/items/mutagen.json:18]:
```suggestion
    "description": "An extremely rare mutagen cocktail, it smells like blood and is a green goop with blood-red streaks",
```

File: data/mods/DinoMod/mutations/mutations.json [hunk 3]
```diff
@@ -0,0 +1,425 @@
...
+      }
+    ],
+    "passive_mods": { "dex_mod": -1 }
+  },
+  {
+    "type": "mutation",
+    "id": "THICKSKIN3",
+    "name": { "str": "Diamond-Skinned" },
+    "points": 4,
+    "description": "Your skin is developing kevlar like qualities.  Makes you slower though.",
```
Comment (actual-nh) [inline data/mods/DinoMod/mutations/mutations.json:117]:
```suggestion
    "description": "Your skin is developing kevlar-like qualities.  Makes you slower though.",
```

File: data/mods/DinoMod/mutations/mutations.json [hunk 4]
```diff
@@ -0,0 +1,425 @@
...
+      }
+    ],
+    "speed_modifier": 0.95
+  },
+  {
+    "type": "mutation",
+    "id": "STRONGBACKDINO",
+    "name": { "str": "Apatosaurus Back" },
+    "points": 2,
+    "description": "You are capable of carrying inhuman amounts more than someone with similar strength could.  Your maximum weight carried is increased by 70%.",
```
Comment (actual-nh) [inline data/mods/DinoMod/mutations/mutations.json:137]:
```suggestion
    "description": "You are capable of carrying inhuman amounts - more than others with similar strength could.  Your maximum weight carried is increased by 70%.",
```

**Review by actual-nh — COMMENTED** [review]
Interesting! IIRC, there is some debate about exactly how cold-blooded dinosaurs were.

**Comment by Maleclypse** [comment]
> Interesting! IIRC, there is some debate about exactly how cold-blooded dinosaurs were.

You are correct but since Raptor already leans into Cold Blooded and I couldn't figure out what the best way to represent the type of not quite hot and not quite cold blooded system that the article I read theorized they may have had, I went with Cold Blooded.

**Comment by actual-nh** [comment]
Medical _**post**_-threshold for the NPCs?

**Comment by actual-nh** [comment]
The general build appears to be erroring in `npc_class::check_consistency()`.

**Comment by Maleclypse** [comment]
> The general build appears to be erroring in `npc_class::check_consistency()` - not sure why...

I put in "bash" instead of "bashing".

**Comment by LyleSY** [comment]
This all looks right to me, but you know mutation modding way better than me. Have you tested this working?

**Comment by Maleclypse** [comment]
> This all looks right to me, but you know mutation modding way better than me. Have you tested this working?

I did but I see I didn't update testing in the PR summary so I'll redo it and double check :)

**Comment by jackledead** [comment]
Fangs in stego trees seem unnecessary. if you're going for a generic herbo feel i say go with beak.

**Comment by Maleclypse** [comment]
> Fangs in stego trees seem unnecessary. if you're going for a generic herbo feel i say go with beak. edit: does the player know its a stego mutation tree? is it called stego in game?

I've gone through this and I can't find Fangs in either tree.  I see where Fangs are a prereq for the Tyrant Teeth.

LyleSY I was able to succesfully inject serum until hitting threshold in both Stego and Tyrant and other tests.

**Pull request merged**